bpf/tests/bpf-prog-test-run/bpf-prog-test-run.go

```diff
@@ -0,0 +1,476 @@
+package main
```

Comment:
Why not make this a testable Go package? (e.g. renaming to `bpf/tests/prog_test.go`)

Reply:
That would require two .go files, right? I was just opting for keeping things simple...

Reply:
Nope, not required:

```
$ ls
foo_test.go
$ go test -v ./...
=== RUN   TestIntMinBasic
--- PASS: TestIntMinBasic (0.00s)
=== RUN   TestIntMinTableDriven
=== RUN   TestIntMinTableDriven/0,1
=== RUN   TestIntMinTableDriven/1,0
=== RUN   TestIntMinTableDriven/2,-2
=== RUN   TestIntMinTableDriven/0,-1
=== RUN   TestIntMinTableDriven/-1,0
--- PASS: TestIntMinTableDriven (0.00s)
    --- PASS: TestIntMinTableDriven/0,1 (0.00s)
    --- PASS: TestIntMinTableDriven/1,0 (0.00s)
    --- PASS: TestIntMinTableDriven/2,-2 (0.00s)
    --- PASS: TestIntMinTableDriven/0,-1 (0.00s)
    --- PASS: TestIntMinTableDriven/-1,0 (0.00s)
PASS
ok      _/home/joe/playground/test      0.003s
```

foo_test.go is a direct copy of the code from https://gobyexample.com/testing .

Reply:
Ah OK, know I understand. @ti-mo 's point was to invoke this as testing packages using `go test`. I think that makes sense.

Reply:
Fixed this (see last patch). 

There is now a `make go_prog_test` target that builds the object file and runs the test. It runs `sudo go test ./tests/prog_test`